[site-isolation] Position:fixed banner at top of imgur.com on iOS flickers when scrolling.

mattwoodrow opened the pull request:
#### b2c98f3202b0f6292ca85cd5f63076cb70d10a3a
<pre>
[site-isolation] Position:fixed banner at top of imgur.com on iOS flickers when scrolling.
<a href="https://bugs.webkit.org/show_bug.cgi?id=287061">https://bugs.webkit.org/show_bug.cgi?id=287061</a>
&lt;<a href="https://rdar.apple.com/142972249">rdar://142972249</a>&gt;

Reviewed by Simon Fraser.

Actual bug fixed by making WKWebViewIOS::_didCommitLayerTree skip most of the
work for subframe process transactions, since they only apply to the main frame.

The main change here is making TransactionID ProcessQualified, so that
transactions from different processes don&apos;t get mixed up in the UI process.

Deletes the existing comparison operators on ProcessQualified (since they
ignored the process), and adds explicit comparison functions that assert if the
process is different.

Used the resulting compilation failures to find all places where we compared
TransactionsIDs and verified that they didn&apos;t try to compare values from
different process. Fixed the ones that did, including the _didCommitLayerTree
code.

ProcessQualified doesn&apos;t have a default constructor, so lots of TransactionID
instances changed to be std::optional.

* Source/WebCore/platform/ProcessQualified.h:
(WebCore::ProcessQualified::generateMonotonic):
(WebCore::ProcessQualified::next const):
(WebCore::ProcessQualified::increment):
(WebCore::ProcessQualified::loggingString const):
(WebCore::ProcessQualified::lessThanSameProcess):
(WebCore::ProcessQualified::lessThanOrEqualSameProcess):
(WebCore::ProcessQualified::greaterThanSameProcess):
(WebCore::ProcessQualified::greaterThanOrEqualSameProcess):
(WebCore::operator&gt;): Deleted.
(WebCore::operator&gt;=): Deleted.
(WebCore::operator&lt;): Deleted.
(WebCore::operator&lt;=): Deleted.
* Source/WebKit/Platform/cocoa/ImageAnalysisUtilities.mm:
(WebKit::requestVisualTranslation):
* Source/WebKit/Scripts/webkit/messages.py:
(serialized_identifiers):
* Source/WebKit/Scripts/webkit/tests/MessageArgumentDescriptions.cpp:
(IPC::serializedIdentifiers):
* Source/WebKit/Shared/MonotonicObjectIdentifier.serialization.in:
* Source/WebKit/Shared/ProcessQualified.serialization.in:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeTransaction.h:
(WebKit::RemoteLayerTreeTransaction::transactionID const):
(WebKit::RemoteLayerTreeTransaction::setTransactionID): Deleted.
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeTransaction.mm:
(WebKit::RemoteLayerTreeTransaction::RemoteLayerTreeTransaction):
(WebKit::dumpChangedLayers):
* Source/WebKit/Shared/TransactionID.h:
* Source/WebKit/Shared/VisibleContentRectUpdateInfo.h:
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _didCommitLoadForMainFrame]):
(-[WKWebView _trackTransactionCommit:]):
(-[WKWebView _restoreScrollAndZoomStateForTransaction:]):
(-[WKWebView _didCommitLayerTree:]):
(-[WKWebView _updateVisibleContentRects]):
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::didCommitLayerTree):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
(WebKit::RemoteLayerTreeDrawingAreaProxy::nextMainFrameLayerTreeTransactionID const):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxy::RemoteLayerTreeDrawingAreaProxy):
(WebKit::RemoteLayerTreeDrawingAreaProxy::commitLayerTreeNotTriggered):
(WebKit::RemoteLayerTreeDrawingAreaProxy::willCommitLayerTree):
(WebKit::RemoteLayerTreeDrawingAreaProxy::commitLayerTreeTransaction):
* Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeDrawingAreaProxyMac.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxyMac::didCommitLayerTree):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxyInternals.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _doubleTapRecognizedForDoubleClick:]):
(-[WKContentView _singleTapRecognized:]):
(-[WKContentView _attemptSyntheticClickAtLocation:modifierFlags:]):
* Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm:
(WebKit::WebPageProxy::resendLastVisibleContentRects):
(WebKit::WebPageProxy::displayedContentScale const):
(WebKit::WebPageProxy::exposedContentRect const):
(WebKit::WebPageProxy::unobscuredContentRect const):
(WebKit::WebPageProxy::inStableState const):
(WebKit::WebPageProxy::unobscuredContentRectRespectingInputViewBounds const):
(WebKit::WebPageProxy::layoutViewportRect const):
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.h:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm:
(WebKit::RemoteLayerTreeDrawingArea::updateRendering):
* Source/WebKit/WebProcess/WebPage/WebFrame.h:
(WebKit::WebFrame::firstLayerTreeTransactionIDAfterDidCommitLoad const):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPageInternals.h:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::attemptSyntheticClick):
(WebKit::WebPage::handleDoubleTapForDoubleClickAtPoint):
(WebKit::WebPage::commitPotentialTap):
(WebKit::WebPage::scaleFromUIProcess const):
(WebKit::WebPage::updateVisibleContentRects):

Canonical link: <a href="https://commits.webkit.org/290132@main">https://commits.webkit.org/290132@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ae80cf3f2ed0cb941fb356c995eb08ef9ae31d37

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/89066 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/8590 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/43819 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/94041 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/39827 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/91117 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/8977 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/16776 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/68624 "Failure limit exceed. At least found 228 new test failures: imported/w3c/web-platform-tests/css/compositing/mix-blend-mode/mix-blend-mode-canvas-parent.html imported/w3c/web-platform-tests/css/compositing/mix-blend-mode/mix-blend-mode-canvas-sibling.html imported/w3c/web-platform-tests/css/compositing/mix-blend-mode/mix-blend-mode-iframe-parent.html imported/w3c/web-platform-tests/css/compositing/mix-blend-mode/mix-blend-mode-iframe-sibling.html imported/w3c/web-platform-tests/css/compositing/mix-blend-mode/mix-blend-mode-plus-lighter-basic.html imported/w3c/web-platform-tests/css/compositing/root-element-background-image-transparency-003.html imported/w3c/web-platform-tests/css/compositing/root-element-filter.html imported/w3c/web-platform-tests/css/css-backgrounds/animations/background-color-animation-with-blur.html imported/w3c/web-platform-tests/css/css-backgrounds/background-repeat/background-repeat-round-roundup.xht imported/w3c/web-platform-tests/css/css-backgrounds/border-radius-clipping-with-transform-001.html ... (failure)") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/26300 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/92068 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/6863 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/80775 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/48986 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/88566 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/6612 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/35165 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/38934 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/76974 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/36132 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/95877 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/16246 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/11885 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/77506 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/16502 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/76577 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/76794 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18929 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/21189 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/19765 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/9338 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/16260 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/21571 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/16001 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/19452 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/17782 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
```diff
@@ -220,8 +219,8 @@ struct PerWebProcessState {
 
-    WebKit::TransactionID firstPaintAfterCommitLoadTransactionID;
-    WebKit::TransactionID lastTransactionID;
+    std::optional<WebKit::TransactionID> resetViewStateAfterTransactionID;
```
Could we use `Markable<>` instead?

mattwoodrow replied:
Unfortunately `MonotonicObjectIdentifier` doesn't support `Markable` (because 0 is a valid value).

It's possible we that could make int max or something (the hash table deleted value) also the markable empty value? Feels like a bit of a hack, but also unlikely to to collide with 64 bits.

smfr commented on Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
```diff
@@ -167,14 +167,14 @@ static bool exceedsRenderTreeSizeSizeThreshold(uint64_t thresholdSize, uint64_t
 #endif
         stopMakingViewBlankDueToLackOfRenderingUpdateIfNecessary();
-        internals().lastVisibleContentRectUpdate = { };
+        internals().lastVisibleContentRectUpdate = std::nullopt;
```
Doesn't the `= { }` still work for optionals?

smfr commented on Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
```diff
@@ -104,8 +104,8 @@ class RemoteLayerTreeDrawingAreaProxy : public DrawingAreaProxy, public RefCount
         CommitLayerTreeMessageState commitLayerTreeMessageState { Idle };
-        TransactionID lastLayerTreeTransactionID;
-        TransactionID pendingLayerTreeTransactionID;
+        std::optional<TransactionID> lastLayerTreeTransactionID;
```
Markable?

smfr commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -10947,7 +10947,7 @@ void WebPageProxy::resetState(ResetStateReason resetStateReason)
     m_scrollPerformanceDataCollectionEnabled = false;
 #endif
-    internals().firstLayerTreeTransactionIdAfterDidCommitLoad = { };
+    internals().firstLayerTreeTransactionIdAfterDidCommitLoad = std::nullopt;
```
Can be left alone?

smfr commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -11019,7 +11019,7 @@ void WebPageProxy::resetState(ResetStateReason resetStateReason)
 
 #if ENABLE(UI_SIDE_COMPOSITING)
-    internals().lastVisibleContentRectUpdate = { };
+    internals().lastVisibleContentRectUpdate = std::nullopt;
```
Ditto

smfr commented on Source/WebKit/UIProcess/WebPageProxyInternals.h:
```diff
@@ -354,7 +354,7 @@ struct WebPageProxy::Internals final : WebPopupMenuProxy::Client
 #if PLATFORM(COCOA)
     WeakObjCPtr<WKWebView> cocoaView;
-    TransactionID firstLayerTreeTransactionIdAfterDidCommitLoad;
+    std::optional<TransactionID> firstLayerTreeTransactionIdAfterDidCommitLoad;
```
Markable?

smfr commented on Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
```diff
@@ -463,7 +463,7 @@ struct ImageAnalysisContextMenuActionData {
 
     CGPoint _lastInteractionLocation;
-    WebKit::TransactionID _layerTreeTransactionIdAtLastInteractionStart;
+    std::optional<WebKit::TransactionID> _layerTreeTransactionIdAtLastInteractionStart;
```
Markable?

smfr commented on Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
```diff
@@ -1593,7 +1593,7 @@ - (void)cleanUpInteraction
     [self removeGestureRecognizer:_touchActionDownSwipeGestureRecognizer.get()];
 
-    _layerTreeTransactionIdAtLastInteractionStart = { };
+    _layerTreeTransactionIdAtLastInteractionStart = std::nullopt;
```
Ditto

smfr commented on Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm:
```diff
@@ -1696,32 +1698,50 @@ static RecommendDesktopClassBrowsingForRequest desktopClassBrowsingRecommendedFo
+    ASSERT(internals().lastVisibleContentRectUpdate);
+    if (internals().lastVisibleContentRectUpdate)
+        return internals().lastVisibleContentRectUpdate->scale();
+    return 0;
```
Probably best to return 1 by default.

smfr commented on Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm:
```diff
@@ -1696,32 +1698,50 @@ static RecommendDesktopClassBrowsingForRequest desktopClassBrowsingRecommendedFo
+    ASSERT(internals().lastVisibleContentRectUpdate);
+    if (internals().lastVisibleContentRectUpdate)
+        return internals().lastVisibleContentRectUpdate->exposedContentRect();
+    return FloatRect();
```
`return { };`

smfr commented on Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm:
```diff
@@ -1696,32 +1698,50 @@ static RecommendDesktopClassBrowsingForRequest desktopClassBrowsingRecommendedFo
 {
-    return internals().lastVisibleContentRectUpdate.unobscuredContentRect();
+    ASSERT(internals().lastVisibleContentRectUpdate);
+    if (internals().lastVisibleContentRectUpdate)
```
Use `value_or` ?

smfr commented on Source/WebKit/WebProcess/WebPage/WebPageInternals.h:
```diff
@@ -41,7 +41,7 @@ struct WebPage::Internals {
     WebCore::VisibleSelection storedSelectionForAccessibility { WebCore::VisibleSelection() };
     FocusedElementInformationIdentifier lastFocusedElementInformationIdentifier;
-    TransactionID lastTransactionIDWithScaleChange;
+    std::optional<TransactionID> lastTransactionIDWithScaleChange;
```
Markable?

mattwoodrow committed:
[site-isolation] Position:fixed banner at top of imgur.com on iOS flickers when scrolling.
https://bugs.webkit.org/show_bug.cgi?id=287061
<rdar://142972249>

Reviewed by Simon Fraser.

Actual bug fixed by making WKWebViewIOS::_didCommitLayerTree skip most of the
work for subframe process transactions, since they only apply to the main frame.

The main change here is making TransactionID ProcessQualified, so that
transactions from different processes don't get mixed up in the UI process.

Deletes the existing comparison operators on ProcessQualified (since they
ignored the process), and adds explicit comparison functions that assert if the
process is different.

Used the resulting compilation failures to find all places where we compared
TransactionsIDs and verified that they didn't try to compare values from
different process. Fixed the ones that did, including the _didCommitLayerTree
code.

ProcessQualified doesn't have a default constructor, so lots of TransactionID
instances changed to be std::optional.

* Source/WebCore/platform/ProcessQualified.h:
(WebCore::ProcessQualified::generateMonotonic):
(WebCore::ProcessQualified::next const):
(WebCore::ProcessQualified::increment):
(WebCore::ProcessQualified::loggingString const):
(WebCore::ProcessQualified::lessThanSameProcess):
(WebCore::ProcessQualified::lessThanOrEqualSameProcess):
(WebCore::ProcessQualified::greaterThanSameProcess):
(WebCore::ProcessQualified::greaterThanOrEqualSameProcess):
(WebCore::operator>): Deleted.
(WebCore::operator>=): Deleted.
(WebCore::operator<): Deleted.
(WebCore::operator<=): Deleted.
* Source/WebKit/Platform/cocoa/ImageAnalysisUtilities.mm:
(WebKit::requestVisualTranslation):
* Source/WebKit/Scripts/webkit/messages.py:
(serialized_identifiers):
* Source/WebKit/Scripts/webkit/tests/MessageArgumentDescriptions.cpp:
(IPC::serializedIdentifiers):
* Source/WebKit/Shared/MonotonicObjectIdentifier.serialization.in:
* Source/WebKit/Shared/ProcessQualified.serialization.in:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeTransaction.h:
(WebKit::RemoteLayerTreeTransaction::transactionID const):
(WebKit::RemoteLayerTreeTransaction::setTransactionID): Deleted.
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeTransaction.mm:
(WebKit::RemoteLayerTreeTransaction::RemoteLayerTreeTransaction):
(WebKit::dumpChangedLayers):
* Source/WebKit/Shared/TransactionID.h:
* Source/WebKit/Shared/VisibleContentRectUpdateInfo.h:
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _didCommitLoadForMainFrame]):
(-[WKWebView _trackTransactionCommit:]):
(-[WKWebView _restoreScrollAndZoomStateForTransaction:]):
(-[WKWebView _didCommitLayerTree:]):
(-[WKWebView _updateVisibleContentRects]):
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::didCommitLayerTree):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
(WebKit::RemoteLayerTreeDrawingAreaProxy::nextMainFrameLayerTreeTransactionID const):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxy::RemoteLayerTreeDrawingAreaProxy):
(WebKit::RemoteLayerTreeDrawingAreaProxy::commitLayerTreeNotTriggered):
(WebKit::RemoteLayerTreeDrawingAreaProxy::willCommitLayerTree):
(WebKit::RemoteLayerTreeDrawingAreaProxy::commitLayerTreeTransaction):
* Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeDrawingAreaProxyMac.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxyMac::didCommitLayerTree):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxyInternals.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _doubleTapRecognizedForDoubleClick:]):
(-[WKContentView _singleTapRecognized:]):
(-[WKContentView _attemptSyntheticClickAtLocation:modifierFlags:]):
* Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm:
(WebKit::WebPageProxy::resendLastVisibleContentRects):
(WebKit::WebPageProxy::displayedContentScale const):
(WebKit::WebPageProxy::exposedContentRect const):
(WebKit::WebPageProxy::unobscuredContentRect const):
(WebKit::WebPageProxy::inStableState const):
(WebKit::WebPageProxy::unobscuredContentRectRespectingInputViewBounds const):
(WebKit::WebPageProxy::layoutViewportRect const):
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.h:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm:
(WebKit::RemoteLayerTreeDrawingArea::updateRendering):
* Source/WebKit/WebProcess/WebPage/WebFrame.h:
(WebKit::WebFrame::firstLayerTreeTransactionIDAfterDidCommitLoad const):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPageInternals.h:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::attemptSyntheticClick):
(WebKit::WebPage::handleDoubleTapForDoubleClickAtPoint):
(WebKit::WebPage::commitPotentialTap):
(WebKit::WebPage::scaleFromUIProcess const):
(WebKit::WebPage::updateVisibleContentRects):

Canonical link: https://commits.webkit.org/290132@main
